pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -268,4 +274,207 @@ public void testMeteredMetrics() {
     testMetrics.getMetricsRegistry().allMetrics().keySet().forEach(testMetrics.getMetricsRegistry()::removeMetric);
     Assert.assertTrue(testMetrics.getMetricsRegistry().allMetrics().isEmpty());
   }
+
+  @Test
+  public void testGlobalGaugeMetricsAsyncAddRemove() throws ExecutionException, InterruptedException {
+    ControllerMetrics controllerMetrics = buildTestMetrics();
+    int gaugeOperationsRuntimeMs = 100;
+
+    ExecutorService executorService = Executors.newFixedThreadPool(2);
+    long endTime = System.currentTimeMillis() + gaugeOperationsRuntimeMs;
+
+    Future<?> addFuture = executorService.submit(() -> {
+      while (System.currentTimeMillis() < endTime + gaugeOperationsRuntimeMs) {
```

Comment:
The idea here is to make `add` calls for higher time than `remove` call and confirm if by the end we have the gauge still available. So `add` is done for 2 * gaugeOperationsRuntimeMs time and remove is `done` for gaugeOperationsRuntimeMs

Reply:
Yeah but this can become flaky. The threads are running independently, and the machines where these tests run are quite overwhelmed most times and there can be delays in thread scheduling, which can lead to the removeGauge to be run last

Reply:
@ankitsultana for synchronous guarantees, I implemented the above with a `CountDownLatch`.